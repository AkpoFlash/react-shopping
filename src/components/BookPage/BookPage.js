import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import find from 'lodash/find';
import PropTypes from 'prop-types';

const Book = styled.div`
	width: 100%;
`;

const Image = styled.header`

`;

const Title = styled.header`

`;

const Author = styled.header`

`;

const Price = styled.header`

`;

const Rating = styled.header`

`;

export const BookPage = (props) => {
	const { id } = props.match.params;
	const book = find( props.books, { 'id': +id } );

	return (
		<Book>
			<Image src={ book.image } />
			<Title>
				{ book.title }
			</Title>
			<Author>
				{ book.author }
			</Author>
			<Price>
				{ book.price }
			</Price>
			<Rating>
				{ book.rating }
			</Rating>
		</Book>
	);
};

const mapStateToProps = ({ books }) => ({
	books: books.items,
});

BookPage.propTypes = {
	books: PropTypes.array.isRequired,
};

export default connect(mapStateToProps)(React.memo(BookPage));