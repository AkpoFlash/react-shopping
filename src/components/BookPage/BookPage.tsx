// Because ts-jest can't do this allowSyntheticDefaultImports
// I created import like this ( * as bla-bla ... )
import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { find } from 'lodash';
import * as PropTypes from 'prop-types';

import { COLOR_BLACK, COLOR_GRAY } from '~/constants/styles';

const Book = styled.div`
	width: 100%;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
`;

const Image = styled.img`
	max-width: 200px;
`;

const Title = styled.h1`
	margin: 10px 0 0;
`;

const Author = styled.h2`
	font-size: 1.2rem;
	color: ${ COLOR_GRAY };
	margin: 10px 0 10px;
`;

const Price = styled.div`
	svg {
    width: 20px;
    height: 20px;
    display: inline-block;
    background: ${ COLOR_BLACK };
  }
`;

const Rating = styled.div`

`;

export const BookPage: React.FunctionComponent<any> = (props) => {
	const { id } = props.match.params;
	const book: any = find( props.books, { 'id': +id } );

	return (
		<Book>
			<Image src={ book.image } />
			<Title>
				{ book.title }
			</Title>
			<Author>
				{ book.author }
			</Author>
			<Price>
				{ book.price }
				<svg>

				</svg>
			</Price>
			<Rating>
				{ book.rating }
			</Rating>
		</Book>
	);
};

const mapStateToProps = ({ books }) => ({
	books: books.items,
});

BookPage.propTypes = {
	books: PropTypes.array.isRequired,
};

export default connect(mapStateToProps)(React.memo(BookPage));