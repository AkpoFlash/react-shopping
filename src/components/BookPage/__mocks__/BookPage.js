export const props = {
	match: {
		params: {
			id: 1,
		}
	},
	books: {
		book: {
			id: 1,
			image: 'default.png',
			title: 'Title book',
			author: 'Author book',
			price: 100,
			rating: 5,
		},
	},
};