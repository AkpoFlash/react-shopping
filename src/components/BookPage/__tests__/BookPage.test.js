import React from 'react';
import { render } from 'enzyme';

import BookPage from '../BookPage';

describe('BookPage', () => {
	const bookPage = render(BookPage);

	it('should BookPage not changed', () => {
		expect(bookPage).toMatchSnapshot();
	});
});
