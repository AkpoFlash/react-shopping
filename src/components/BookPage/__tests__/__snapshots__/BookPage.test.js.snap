// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dumb BookPage check BookPage snapshot 1`] = `
<styled.div>
  <styled.img
    src="default.png"
  />
  <styled.h1>
    Title book
  </styled.h1>
  <styled.h2>
    Author book
  </styled.h2>
  <styled.div>
    100
    <svg />
  </styled.div>
  <styled.div>
    5
  </styled.div>
</styled.div>
`;
